Try to unify how documentation is shown in all platforms (#122)
undefinedlabs/scope-docs: docs/java-configuration.md

sidebar_label: Configuration
---

> The behaviour of the Scope Agent can be modified by using  environment variables or by using a configuration file `Scope.yml`in the source root of the project.

Author: Changed

> If these properties are manually configured, they will be `true` only on encountering the string `true` configured on the environment variable. Any other value will be considered as `false`.

## Disabling Scope instrumentation

You can disable Scope auto instrumentation. Scope auto instruments your code and tests by default, you can disable this behaviour.

### Using Environment Variables

```sh
SCOPE_AUTO_INSTRUMENT=false
```

### Using YAML Configuration file

```yaml
scope:

>   auto_instrument: false

Reviewer: In Java is:
```yaml
scope:
  instrumentation:
    enabled: true
```

Reviewer 2: as in dotnet, so the env-var is: `SCOPE_INSTRUMENTATION_ENABLED `

Author: changed

  
```

## Changing service name

You can change the name of the service when sending data to Scope. Scope initially uses `default` for the service name.

### Using Environment Variables

```sh

> SCOPE_SERVICE="service name"

Reviewer: ```suggestion
SCOPE_SERVICE="service-name"
```

Author: changed

```

### Using YAML Configuration file

```yaml
scope:

>   service: 'service name'

Reviewer: ```suggestion
  service: 'service-name'
```

Author: changed

```

## Setting Scope as Global Tracer

> You can set Scope as the Global Tracer of Open Tracing. Scope uses Open Tracing internally to handle Span and traces, using its own tracer, You can set it as the global tracer so it will capture all the spans generated by any Open Tracing library in relation to your tests. 

Reviewer: ```suggestion
If you set Scope as OpenTracing Global Tracer, your own spans will be captured and shown as part of the Scope trace view for a certain test.
```

Author: Changed in all agents

### Using Environment Variables

```sh

> SCOPE_SET_GLOBAL_TRACER=true

Reviewer 2: The env var should be: `SCOPE_TRACER_GLOBAL` to follow the `scope.yml` pattern:
```yaml
scope:
    tracer:
        global: true
```

Author: changed for .net only

Author: Added an entry for next agents guild meeting

```

### Using YAML Configuration file

```yaml
scope:

>   set_global_tracer: true

Reviewer: Now in Java is:
```yaml
scope:
  tracer:
    global: true
```
Should Java agent change?

  
```

> ## Change testing mode

Reviewer: ```suggestion
## Set/Unset testing mode manually
```

These variable indicates to the agent if it is running tests (`true`) or it is being used for runtime instrumentation (`false`). By default it will run in testing mode

### Using Environment Variables

```sh
SCOPE_TESTING_MODE=false
```

### Using YAML Configuration file

```yaml
scope:

>   testing_mode: false

Reviewer: Right now in Java is:
```yaml
scope:
  mode:
    testing: true
```

Should Java agent change it?

Author: Changed in all docs

>   testing_mode: false

Reviewer: Should we document this flag? It looks like an implementation detail for me.

Author: Added to next guild meeting agenda

  
```

>  By default `SCOPE_TESTING_MODE` property depends on whether the build has been triggered by a CI server (`true`), or not (`false`).

Reviewer: ```suggestion
If `SCOPE_TESTING_MODE` is not specified, the agent sets to  `true` when running in CI and `false` otherwise.
```

Author: changed for all agents

sidebar_label: Configuration
---

The behaviour of the Scope Agent can be modified using environment variables or a configuration file `scope.yml` in the source root of the project.

> > If these properties are manually configured, they will be `true` only on encountering the string `true` configured on the environment variable. Any other value will be considered as `false`.

Author: changed for all

## Disabling Scope instrumentation

Scope auto instruments your code and tests by default. You can disable this behaviour.

### Using Environment Variables

```sh

> SCOPE_AUTO_INSTRUMENT=false

Reviewer 3: This should be 
```suggestion
SCOPE_INSTRUMENTATION_ENABLED=false
```
for consistency with other agents, right? @drodriguezhdez

```

### Using YAML Configuration file

```yaml
scope:

>   scope:

Reviewer 3: ```suggestion
```

  instrumentation:
    enabled: false
```

## Changing service name

You can specify the name of the service when sending data to Scope. Scope uses `default` as fallback.

### Using Environment Variables

```sh
SCOPE_SERVICE="service-name"
```

### Using YAML Configuration file

```yaml
scope:
  service: 'service-name'
```

## Setting Scope as Global Tracer

If you set Scope as OpenTracing Global Tracer, your own spans will be captured and shown as part of the Scope trace view for a certain test.

### Using Environment Variables

```sh
SCOPE_SET_GLOBAL_TRACER=true
```

### Using YAML Configuration file

```yaml
scope:
  tracer:
    global: true

>   

Reviewer 3: ```suggestion
```

```

## Change testing mode

This variable indicates whether the agent is running tests (`true`) or it is being used for runtime instrumentation (`false`). If `SCOPE_TESTING_MODE` is not set, the agent sets to `true` when testing frameworks are present in classpath. Otherwise, the agent sets to `false`.

### Using Environment Variables

```sh
SCOPE_TESTING_MODE=false
```

### Using YAML Configuration file

```yaml
scope:
  mode:

>     testing: false

Reviewer 3: Same comment as with .NET

>   

Reviewer 3: ```suggestion
```
